hyperspy/io_plugins/jeol.py

```diff
@@ -267,8 +319,45 @@ def check_multiple(factor, number, string):
             channel_number = int(
                 np.round((cutoff_at_kV - energy_offset) / energy_scale)
                 )
+        # pixel time in milliseconds
+        pixel_time = meas_data_header["Doc"]["DwellTime(msec)"]
 
-        axes = [
+        data_shape = [height, width, channel_number]
+        # Sweep value is not reliable
+        # sweep = meas_data_header["Doc"]["Sweep"]
+        frame_ptr_list, last_valid, has_em_image = pts_prescan(rawdata, width, height)
+        if last_valid or not only_valid_data:
+            sweep = len(frame_ptr_list) - 1
+        else:
+            sweep = len(frame_ptr_list) - 2
+        if not frame_list:
+            frame_list = range(sweep)
+        frame_list = list(frame_list)
+        frame_list2 = []
+        for frame_idx in frame_list:
+            if frame_idx < 0 or frame_idx >= sweep:
+                _logger.info(f"Ignoreing frame {frame_idx} : Selected frame is not found in pts data.")
+                continue
+            frame_list2.append(frame_idx)
+        frame_list = np.array(frame_list2)
```

Comment:
It is a bit suggestive, I find the following more clear:
```suggestion
        if not frame_list:
            frame_list = np.arange(sweep)
        frame_list = np.asarray(frame_list)
        wrong_frames_idx = np.argwhere((frame_list<0) | (frame_list>=sweep))
        if len(wrong_frames_idx) > 0:
            wrong_frames = frame_list[wrong_frames_idx].flatten().tolist()
            _logger.info(f"Ignoring frame {wrong_frames} : Selected frame is not found in pts data.")
```